src/shared/pages/user/userAdd/__tests__/addUser.spec.tsx

```diff
@@ -0,0 +1,174 @@
+import AddUser from '../addUser';
+import * as UsersService from '../../../../services/usersService';
+import { ADD_USER_BAD_REQUEST_TITLE, ADD_USER_ERROR_DESCRIPTION, ADD_USER_ERROR_TITLE } from '../../../../models/constants';
+import { User } from '../../../../models/user';
+import * as useError from '../../../../hooks/useError';
+
+let match: match<any>;
+let history: History<any>;
+let location: Location;
+let mockUser: User;
+let wrapper: RenderResult;
+
+const useReducerSpy = jest.spyOn(React, 'useReducer');
+const reducerDispatch = jest.fn();
+const useErrorSpy = jest.spyOn(useError, 'default');
+const addFormErrorSpy = jest.fn();
+const clearErrorsSpy = jest.fn();
+const setMessageSpy = jest.fn();
+
+const userId = 'xx-xx-xx-xx';
+
+const renderComponent = () => render(
+    <MemoryRouter>
+        <AddUser history={history} location={location} match={match}/>
+    </MemoryRouter>
+);
+
+jest.spyOn(UsersService, 'addUser').mockImplementation(() => Promise.resolve(userId));
+
+beforeEach(() => {
+    history = createBrowserHistory();
+    match = {
+        isExact: true,
+        params: {},
+        path: '',
+        url: ''
+    };
+
+    location = {
+        hash: '',
+        key: '',
+        pathname: '',
+        search: '',
+        state: {}
+    };
+    mockUser = {
+        id: userId,
+        username: 'un',
+        email: '',
+        firstName: '',
+        lastName: '',
+        enabled: true
+    };
+    useReducerSpy.mockImplementation(() => [mockUser, reducerDispatch]);
+    useErrorSpy.mockImplementation(() => [{}, addFormErrorSpy, clearErrorsSpy, setMessageSpy]);
+    history.push = jest.fn();
+    reducerDispatch.mockReset();
+    addFormErrorSpy.mockReset();
+    clearErrorsSpy.mockReset();
+    setMessageSpy.mockReset();
+    act(() => {
+        wrapper = renderComponent();
+    });
+});
+
+describe('when the addUser component is mounted', () => {
+    it('should render with default props', async () => {
+        expect.assertions(1);
+
+        await wait(() => {
+            expect(wrapper.container).toMatchSnapshot();
+        });
+    });
+});
+
+describe('when the submit button is clicked', () => {
+    describe('and the data is filled in', () => {
+
+        beforeEach(async () => {
+            mockUser.email = 'firstname@test.com';
+            mockUser.firstName = 'firstname';
+            mockUser.lastName = 'lastname';
+            const submitButton = await waitForElement(async () => {
+                return await wrapper.findByText('Submit');
+            });
+
+            fireEvent.click(submitButton);
+        });
+
+        describe('and the service call is successful', () => {
+            it('should redirect to the view user page', async () => {
+                expect.assertions(1);
+
+                await wait(() => {
+                    // history.push(`/user-view/${userUUID}`, { successMessage: constants.ADD_USER_SUCCESS });
```

Comment:
Why is this commented out?

Reply:
I have now removed this line. I added it temporarily while working on this test.